cloudgenix_config/pull.py

```diff
@@ -1263,9 +1280,15 @@ def _pull_config_for_single_site(site_name_id):
             ntp_template = copy.deepcopy(ntp)
             strip_meta_attributes(ntp_template, leave_name=True)
             if ntp.get('source_interface_ids'):
-                source_ids = []
+                source_ids, bps = [], ''
                 for iface in ntp.get('source_interface_ids', []):
+                    # Add 'parent_type' field if model is 9k and interface is bypasspair in 12-16
+                    if ion_9k and if_id2type.get(iface) == 'bypasspair' and int(id_name_cache.get(iface)) in range(12,17):
+                        bps += '_' + id_name_cache.get(iface, iface)
+                        ntp_template['parent_type'] = if_id2type[iface]
                     source_ids.append(id_name_cache.get(iface, iface))
+                if bps:
+                    ntp_template['parent_type'] = 'bypasspair' + bps
```

Comment:
Why is the interface name appended to parent_type?

Reply:
ntps can contain multiple bps. Thats the reason to append it that way. It will be 'bypasspair_12_56' when there are 2 bps